services/placement/service/placementservice.go

```diff
@@ -161,15 +188,68 @@ func (ps placementService) ReplaceInstance(
 		return nil, err
 	}
 
-	if p, err = ps.algo.ReplaceInstance(p, leavingInstance, addingInstances); err != nil {
+	if p, err = ps.algo.ReplaceInstance(p, leavingInstanceID, addingInstances); err != nil {
 		return nil, err
 	}
 
-	return p, ps.validateAndSaveSnapshot(p)
+	if err := placement.Validate(p); err != nil {
+		return nil, err
+	}
+
+	// TODO(chaowant) this will be removed once the m3db nodes start to mark shards as available
+	p, err = markAllShardsAsAvailable(p)
```

Comment:
This too, on replace the nodes will mark the shards as available once initialized.